### DISCUSSION TITLE
Elevation Gain Offset Setting

### DISCUSSION THREAD

**Discussion by hellobensalt** [cagnulein/qdomyos-zwift#2260] [Ideas]
_I'm not sure if this is the correct place for this? But, if not, please let me know where is._

Since a 1-2% incline on a treadmill is widely considered to be equivalent to the energy costs of running outside on the flat with air resistance and with full use of hamstrings (i.e, no assistance from the spinning belt), I'd like to be able to ignore my baseline incline on all runs. I don't consider this 1-2% to be equivalent to real-world elevation gain.

So, I would want QZ to ignore my 2% baseline incline (i.e., treat it as 0 metres elevation gain). For example:

1) Zwift at 0% means treadmill at 2% and 0 m gain per 1 km run
2) Zwift at 1% means treadmill at 3% and +10 m gain per 1 km run
3) Zwift at 2% means treadmill at 4% and +20 m gain per 1 km run
4) Zwift at -1% means treadmill at 1% and 0 m gain per 1 km run (or -10 m loss per 1 km run)

Given the current setting options, would a further setting to offset the elevation gain work? So, I could set incline offset to 2, min incline to -100 and an additional elevation gain offset to -2?

Many thanks for your help - as always.

**Comment by cagnulein** [comment]
I will look into this and get back to you.


Il giorno gio 28 mar 2024 alle 11:45 hellobensalt ***@***.***>
ha scritto:

**Reply by hellobensalt** [reply]
@cagnulein Thanks, again.

Now I'm becoming more familiar with the QZ settings, I realise the incline offset is in the Zwift settings.

So, what if I wanted to run a structured workout/free run with the 2% incline and also have that baseline ignored there too? I could set up a workout to add the desired elevation gain to that 2% (so, 5% gain would be 7% on my treadmill). But, the elevation gain offset would need to apply to both Zwift and a free run.

I also see incline override settings for horizon treadmills, but don't know if that's relevant here?

**Reply by cagnulein** [reply]
mmm the inclination overrides could work for you but I didn't implement this for the horizon but I could certainly do! So decrasing the real inclination by 2% in the table should give you exactly what you need!

so 0% will become -2%
1% will become -1 %
2% will become 0%

and so on

Do you agree?

**Reply by hellobensalt** [reply]
I'm not sure. There are three key things related to this that need to work with my 2% baseline incline:

1) Zwift run - an elevation offset to ignore my 2% baseline incline. This can currently be done in the QZ Zwift settings by setting the incline offset to 2 and min incline to -100 (the default)

2) Free run - an elevation offset to ignore my 2% baseline incline. In the QZ Horizon treadmill settings, I can see the incline override settings we referred to above. But you say it's not been implemented for Horizon treadmills? Anyhow, would setting this to -2, as you describe, work for free runs and for Zwift runs? Or would I still need to set the incline offset for Zwift to 2?

3) An elevation gain offset - so, the total elevation gain of a run ignores my 2% incline. Only elevation above that 2% is considered elevation gain.

I hope this is clear?  And a big thank you - again!

**Comment by cagnulein** [comment]
let me try to simplify this: with the inclination overrides you can remap
each inclination values from the treadmill so you can virtually do
everything. it will be simpler to test this. I will implement this after
Easter. Happy easter!
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 30 mar 2024 alle ore 08:35 hellobensalt <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
@hellobensalt done!
https://github.com/cagnulein/qdomyos-zwift/actions/runs/8495558616
Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

if you are instead on ios I can send you a new build. What you have to do with this build is to use only the inclination overrides, in this way all the requests from zwift will be transparent and they will use the inclination overrides table!
Let m know

**Reply by hellobensalt** [reply]
Sorry for the delay in getting back to you, @cagnulein! And a belated Happy Easter.

I'll test all this over the next few days and will report back. (Btw, I'm on Android.)

Many thanks.

**Reply by hellobensalt** [reply]
Sorry, @cagnulein. I'm relatively unfamiliar with Github. Where exactly is the APK I can download?

**Reply by cagnulein** [reply]
here https://github.com/cagnulein/qdomyos-zwift/actions/runs/8495558616/artifacts/1372182051

**Reply by cagnulein** [reply]
here https://github.com/cagnulein/qdomyos-zwift/actions/runs/8495558616/artifacts/1372182051

**Reply by hellobensalt** [reply]
Thank you!

**Reply by hellobensalt** [reply]
I'm about to test this, but before I do I wanted to check this:

Do I need to set the Inclination Override Offset to 2% or do I need to individually change every single incline percentage in the table from 0 - 15%?

Thanks again for your help.

**Reply by hellobensalt** [reply]
Okay, I've just tested it with the Inclination Override Offset set to -2. That seems to work with manual controls and what QZ sees as the inclination of my treadmill. But I need to test it with Zwift.

One issue I'm having with QZ is after I click on stop, none of the charts on the completed activity screen contain any data. Just axis without any data points. Also, none of these test activities are uploaded to Strava. So, I'm not sure how to do a more thorough test?

Is it because I've only been using the 15 min trial versions? 

But unless I can see the data that QZ records from these test activities, I can't test to see if it's all working correctly.

**Reply by cagnulein** [reply]
The chart thing and strava thing it's because it's a CI build from github. I pushed today the update on the play store too so if it works with zwift you will have a 100% working version on the play store soon
